packages/xstate-wallet/src/utils.ts

```diff
@@ -0,0 +1,36 @@
+import {filter, map, flatMap, concatMap} from 'rxjs/operators';
+import {from, OperatorFunction} from 'rxjs';
+
+function flatFilter<T>(predicate: (event: T) => boolean | Promise<boolean>) {
```

Comment:
It might be nice to isolate observable utils in `rxjs-utils` or something like that.